src/udev/udev-builtin-path_id.c

```diff
@@ -669,6 +681,14 @@ static int builtin_path_id(sd_device *dev, int argc, char *argv[], bool test) {
 
                 udev_builtin_add_property(dev, test, "ID_PATH", path);
                 udev_builtin_add_property(dev, test, "ID_PATH_TAG", tag);
+                /* compatible link for ATA devices */
+                if ((p=strstr(path, "ata-")) != NULL) {
+                    const char *del_str;
+                    char compat_link[UTIL_NAME_SIZE];
+                    del_str = strchr(p, '.');
+                    strncpy_exact(compat_link, path, strlen(path)-strlen(del_str));
+                    udev_builtin_add_property(dev, test, "ID_PATH_ATA_COMPAT", compat_link);
+                }
```

Comment:
coding style is not OK. Please see:

https://systemd.io/CODING_STYLE/

at the most basic, the indentation level is all off... this should have comments, explaining what you are doing here... and subtracting the string lengths is weird... really weird, if you just want to copy out a prefix... just do pointer difference to determine the right length

Reply:
Thanks for the review. I have updated the source accordingly.